Tropos/Sources/Controllers/TRApplicationController.m

```diff
@@ -79,21 +79,31 @@ - (void)setMinimumBackgroundFetchIntervalForApplication:(UIApplication *)applica
 - (void)subscribeToNotificationsWithDeviceToken:(NSData *)deviceToken
 {
     NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
-    NSString *formerChannelKey = @"CourierChannel";
+    NSString *courierChannelKey = @"CourierChannel";
 
     NSTimeZone *timeZone = [NSTimeZone localTimeZone];
     NSString *channel = [TRCourierClient channelNameForTimeZone:timeZone];
 
-    NSString *formerChannel = [userDefaults stringForKey:formerChannelKey];
+    NSString *formerChannel = [userDefaults stringForKey:courierChannelKey];
     if (formerChannel && ![channel isEqualToString:formerChannel]) {
         NSLog(@"unsubscribe from channel: %@", formerChannel);
-        [self.courier unsubscribeFromChannel:formerChannel];
+        [[RACSignal createSignal:^RACDisposable *(id<RACSubscriber> subscriber) {
+            [self.courier unsubscribeFromChannel:formerChannel completionHandler:^(NSError * _Nullable error) {
+                if (!error) {
+                    [subscriber sendCompleted];
+                } else {
+                    [subscriber sendError:error];
+                }
+            }];
+
+            return [RACDisposable disposableWithBlock:^{}];
+        }] retry];
```

Comment:
Because you're not actually using the signal returned from `-retry`, it looks like this code might never actually be running...